### PR TITLE
Vectorize `make_vector`

### PR DESCRIPTION
**Pull request by ricardoV94**
## Description

Showed up in some vectorized logps in https://github.com/pymc-devs/pymc-experimental/pull/353

We can always convert a vectorized MakeVector (stack of scalars) into a join (concatenate) of tensors. May just need to explicitly broadcast them first. No good reason not to do it eagerly

## Related Issue

- [x] Related to #430 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by ricardoV94** [comment]
I should test it doesn't fail when all inputs are still scalars or just one of them

File: pytensor/tensor/basic.py [hunk 1]
```diff
@@ -1888,6 +1888,25 @@ def _get_vector_length_MakeVector(op, var):
...
+    if not all(
+        batch_input.type.broadcastable == bcast_pattern for batch_input in batch_inputs
+    ):
+        batch_inputs = broadcast_arrays(*batch_inputs)
+
+    # Join along the last axis
+    new_out = join(
+        -1, *[expand_dims(batch_inputs, axis=-1) for batch_inputs in batch_inputs]
+    )
+    return new_out.owner
```
Comment (jessegrabowski) [inline pytensor/tensor/basic.py:1907]: why `.owner` here?

**Review by jessegrabowski — APPROVED** [review]

File: pytensor/tensor/basic.py (hunk 1 above)
Comment (ricardoV94) [reply pytensor/tensor/basic.py:1907]: `vectorize_node` returns a node. No idea why, could probably have made it like rewrites and return the output variables

Comment (jessegrabowski) [reply pytensor/tensor/basic.py:1907]: I guess it's right there in the name eh?

Comment (ricardoV94) [reply pytensor/tensor/basic.py:1907]: It's silly and potentially too restrictive, opened an issue: https://github.com/pymc-devs/pytensor/issues/902

**Pull request merged**